pkg/authorization/path/path.go

```diff
@@ -18,71 +18,46 @@ package path
 
 import (
 	"context"
-	"path"
-	"strings"
+	"fmt"
 
-	"k8s.io/apimachinery/pkg/util/sets"
 	"k8s.io/apiserver/pkg/authorization/authorizer"
+	pathauthorizer "k8s.io/apiserver/pkg/authorization/path"
 )
 
 type pathAuthorizer struct {
-	matchDecision, noMatchDecision authorizer.Decision
-
-	paths        sets.String
-	pathPatterns []string
-}
-
-func newPathAuthorizer(onMatch, onNoMatch authorizer.Decision, inputPaths []string) *pathAuthorizer {
-	var patterns []string
-	paths := sets.NewString() // faster than trying to match every pattern every time
-	for _, p := range inputPaths {
-		p = strings.TrimPrefix(p, "/")
-		if len(p) == 0 {
-			// matches "/"
-			paths.Insert(p)
-			continue
-		}
-		if strings.ContainsRune(p, '*') {
-			patterns = append(patterns, p)
-		} else {
-			paths.Insert(p)
-		}
-	}
-
-	return &pathAuthorizer{
-		matchDecision:   onMatch,
-		noMatchDecision: onNoMatch,
-		paths:           paths,
-		pathPatterns:    patterns,
-	}
+	delegatedPathAuthorizer authorizer.Authorizer
 }
 
 func (a *pathAuthorizer) Authorize(ctx context.Context, attr authorizer.Attributes) (authorizer.Decision, string, error) {
-	pth := strings.TrimPrefix(attr.GetPath(), "/")
-	if a.paths.Has(pth) {
-		return a.matchDecision, "", nil
-	}
+	decision, reason, err := a.delegatedPathAuthorizer.Authorize(ctx, attr)
 
-	for _, pattern := range a.pathPatterns {
-		if found, err := path.Match(pattern, pth); err != nil {
-			return authorizer.DecisionNoOpinion, "Error", err
-		} else if found {
-			return a.matchDecision, "", nil
-		}
+	// There is a match on the path, so we have no opinion and let subsequent
+	// authorizers in a union decide.
+	if err == nil && decision == authorizer.DecisionAllow {
+		return authorizer.DecisionNoOpinion, reason, nil
 	}
 
-	return a.noMatchDecision, "", nil
+	return authorizer.DecisionDeny, fmt.Sprintf("NOT(%s)", reason), err
 }
 
-func NewAllowPathAuthorizer(allowPaths []string) authorizer.Authorizer {
-	if len(allowPaths) == 0 {
-		return authorizer.AuthorizerFunc(func(context.Context, authorizer.Attributes) (authorizer.Decision, string, error) {
-			return authorizer.DecisionNoOpinion, "", nil
-		})
+// NewAllowedPathsAuthorizer returns an authorizer that has no opinion on
+// matches for the given paths and denies all others. No opinion ensures that
+// subsequent authorizers in a union are still called. The given paths can
+// include simple glob patterns.
```

Comment:
```suggestion
// NewAllowedPathsAuthorizer returns an authorizer that allows requests to
// `allowPaths` through with `NoOpinion` and denies all others. This ensures that
// subsequent authorizers in a union are still called on the requests that pass through
// this authorizer.
// The provided paths can include simple glob patterns.
```
What's simple glob patterns?